### ISSUE TITLE
Error while copying database (pma__column_info)

### ISSUE THREAD

**Issue by yeezy69** [phpmyadmin/phpmyadmin#14447]
I heavy use the phpmyadmin "Internal relation" function.

While copying (structure and data) a hole database, it's gives

`1136 - Column count doesn't match value count at row 1`

error about the sql-query:

`REPLACE INTO phpmyadmin.pma__column_info (db_name, table_name, column_name, comment, mimetype, transformation, transformation_options)  VALUES('newdatabase','newtable','fieldabc','ListeA','','','','')`

I see 7 fields and 8 values in this query.

Affected versions i tried: phpMyAdmin-4.7.9 and phpMyAdmin-4.8.2
And not affected version for me is: phpMyAdmin-4.6.6

My "phpmyadmin" database is actual.

Apache/2.4.18
Database-Client Version: libmysql - mysqlnd 5.0.12-dev - 20150407
PHP-Extension: mysqli curl mbstring
PHP-Version: 7.0.30-0ubuntu0.16.04.1
Server-Type: MariaDB
Server-Version: 10.0.34-MariaDB-0ubuntu0.16.04.1 - Ubuntu 16.04
Protocol-Version: 10

**Comment by williamdes** [comment]
Did you upgrade phpmyadmin database ? https://github.com/phpmyadmin/phpmyadmin/tree/master/sql

TLDR; try creating a new phpmyadmin database.

**Comment by yeezy69** [comment]
my phpmyadmin database is actual. i checked the structure manually.

i browsed in code:

`phpMyAdmin-4.6.6-all-languages/libraries/Table.php:1146

            // Write every comment as new copied entry. [MIME]
            while ($comments_copy_row
                = $GLOBALS['dbi']->fetchAssoc($comments_copy_rs)) {
                $new_comment_query = 'REPLACE INTO '
                    . Util::backquote($GLOBALS['cfgRelation']['db'])
                    . '.' . Util::backquote(
                        $GLOBALS['cfgRelation']['column_info']
                    )
                    . ' (db_name, table_name, column_name, comment'
                    . ($GLOBALS['cfgRelation']['mimework']
                        ? ', mimetype, transformation, transformation_options'
                        : '')
                    . ') ' . ' VALUES(' . '\'' . $GLOBALS['dbi']->escapeString($target_db)
                    . '\',\'' . $GLOBALS['dbi']->escapeString($target_table) . '\',\''
                    . $GLOBALS['dbi']->escapeString($comments_copy_row['column_name'])
                    . '\''
                    . ($GLOBALS['cfgRelation']['mimework']
                        ? ',\'' . $GLOBALS['dbi']->escapeString(
                            $comments_copy_row['comment']
                        )
                        . '\',' . '\'' . $GLOBALS['dbi']->escapeString(
                            $comments_copy_row['mimetype']
                        )
                        . '\',' . '\'' . $GLOBALS['dbi']->escapeString(
                            $comments_copy_row['transformation']
                        )
                        . '\',' . '\'' . $GLOBALS['dbi']->escapeString(
                            $comments_copy_row['transformation_options']
                        )
                        . '\''
                        : '')
                    . ')';
                PMA_queryAsControlUser($new_comment_query);
            } // end while
`

and

`phpMyAdmin-4.8.2-all-languages/libraries/classes/Table.php:1239

            // Write every comment as new copied entry. [MIME]
            while ($comments_copy_row
                = $GLOBALS['dbi']->fetchAssoc($comments_copy_rs)) {
                $new_comment_query = 'REPLACE INTO '
                    . Util::backquote($GLOBALS['cfgRelation']['db'])
                    . '.' . Util::backquote(
                        $GLOBALS['cfgRelation']['column_info']
                    )
                    . ' (db_name, table_name, column_name, comment'
                    . ($GLOBALS['cfgRelation']['mimework']
                        ? ', mimetype, transformation, transformation_options'
                        : '')
                    . ') ' . ' VALUES(' . '\'' . $GLOBALS['dbi']->escapeString($target_db)
                    . '\',\'' . $GLOBALS['dbi']->escapeString($target_table) . '\',\''
                    . $GLOBALS['dbi']->escapeString($comments_copy_row['column_name'])
                    . '\',\'' . $GLOBALS['dbi']->escapeString($target_table) . '\',\''
                    . $GLOBALS['dbi']->escapeString($comments_copy_row['comment'])
                    . '\''
                    . ($GLOBALS['cfgRelation']['mimework']
                        ? ',\'' . $GLOBALS['dbi']->escapeString(
                            $comments_copy_row['mimetype']
                        )
                        . '\',' . '\'' . $GLOBALS['dbi']->escapeString(
                            $comments_copy_row['transformation']
                        )
                        . '\',' . '\'' . $GLOBALS['dbi']->escapeString(
                            $comments_copy_row['transformation_options']
                        )
                        . '\''
                        : '')
                    . ')';
                $relation->queryAsControlUser($new_comment_query);
            } // end while
            `

in version of 4.8.2 there is two times
`. '\',\'' . $GLOBALS['dbi']->escapeString($target_table) . '\',\''`

I think this is the error. Because of this query with 2 times newtable:

`VALUES('newdatabase','newtable','fieldabc','newtable','','','','')`

**Comment by yeezy69** [comment]
Seems to work for me

```diff -Nau /tmp/Table.php phpMyAdmin-4.8.2-all-languages/libraries/classes/Table.php 
--- /tmp/Table.php	2018-07-04 00:08:19.551608016 +0200
+++ phpMyAdmin-4.8.2-all-languages/libraries/classes/Table.php	2018-07-04 00:07:40.939143990 +0200
@@ -1251,7 +1251,7 @@
                     . ') ' . ' VALUES(' . '\'' . $GLOBALS['dbi']->escapeString($target_db)
                     . '\',\'' . $GLOBALS['dbi']->escapeString($target_table) . '\',\''
                     . $GLOBALS['dbi']->escapeString($comments_copy_row['column_name'])
-                    . '\',\'' . $GLOBALS['dbi']->escapeString($target_table) . '\',\''                    
+                    . '\',\''
                     . $GLOBALS['dbi']->escapeString($comments_copy_row['comment'])
                     . '\''
                     . ($GLOBALS['cfgRelation']['mimework']`

**Comment by williamdes** [comment]
@yeezy69 I can not reproduce this issue, open a pull-request to propose your patch :)

**Comment by ibennetch** [comment]
I was also unable to reproduce this issue, although your analysis and explanation seem to make sense to me.

c4654d24dfb is when this was last changed, to fix #13010, which related to an incorrect column count when performing a copy operation. It's a bit tough for me to follow off hand what the intention is there, but I agree with you that it shouldn't include $target_table twice.

**Comment by ThallerIT** [comment]
thanks for the patch, @yeezy69 - well done!
